src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -119,6 +125,83 @@ public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Ex
         }
     }
 
+    abstract void doBatchExecute(List<String> inferenceList, Consumer<List<?>> handler, Consumer<Exception> onException);
```

Comment:
Do we need to implement a sync-style doBatchExecute here like the `execute(IngestDocument ingestDocument)` and `execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Exception> handler)`

Reply:
We didn't create the version without callback of `batchExecute` in `Processor` as it's only the callback version which is triggered from pipeline.